src/kalturaMux.js

```diff
@@ -71,31 +81,81 @@ const initKalturaMux = function (player, options) {
 
   // Allow mux to automatically retrieve state information about the player on each event sent
   options.getStateData = () => {
-    const videoElement = player.getVideoElement();
+    // NOTE: getVideoElementProps and getPlayerProps provide "safeCall" util with a function to get
+    // the props instead just accessing the prop. This is needed so "safeCall" util can run properly
+    // because it needs a function as parameter.
+    // For example: the sourceurl comes in player.selectedSource.url but selectedSource can not be
+    // passed to safeCall since it is not a function. If it was player.selectedSource().url then it would work.
+
+    let getVideoElementProps = (propName) => {
+      const videoElement = player.getVideoElement();
+      let props = {
+        width: getComputedStyle(videoElement, 'width'),
+        height: getComputedStyle(videoElement, 'height')
+      };
+
+      if (videoElement.videoHeight) {
```

Comment:
This is causing an error for me, should just need an extra null check.
```suggestion
      if (videoElement && videoElement.videoHeight) {
```